.github/workflows/eamxx-kokkos-develop-testing.yml

```diff
@@ -0,0 +1,159 @@
+name: eamxx-sa-kokkos-develop
+
+on:
+  # Manual run is used to bless
+  workflow_dispatch:
+    inputs:
+      job_to_run:
+        description: 'Job to run'
+        required: true
+        type: choice
+        options:
+          - gcc-openmp
+          - gcc-cuda
+          - all
+      bless:
+        description: 'Generate baselines'
+        required: true
+        type: boolean
+      submit:
+        description: 'Force cdash submission'
+        required: true
+        type: boolean
+
+  # Add schedule trigger for nightly runs at midnight MT (Standard Time)
+  schedule:
+    - cron: '0 7 * * 1'  # Runs at 7 AM UTC on Mondays, which is midnight MT during Standard Time
+
+concurrency:
+  # Two runs are in the same group if they are testing the same git ref
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  # Submit to cdash only for nightlies or if the user explicitly forced a submission via workflow dispatch
+  submit: ${{ github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && inputs.submit) }}
+  generate: ${{ github.event_name == 'workflow_dispatch' && inputs.bless }}
+  # Do EKAT testing for nightlies or (TODO: if EKAT label is present)
+  ekat: ${{ github.event_name == 'schedule' }}
+
+jobs:
+  gcc-openmp:
+    if: |
+      ${{
+        github.event_name != 'workflow_dispatch' ||
+        (
+          github.event.inputs.job_to_run == 'gcc-openmp' ||
+          github.event.inputs.job_to_run == 'all'
+        )
+      }}
+    runs-on:  [self-hosted, ghci-snl-cpu, gcc]
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type: [sp, dbg, fpe, opt]
+    name: gcc-openmp / ${{ matrix.build_type }}
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v5
+        with:
+          persist-credentials: false
+          show-progress: false
+          submodules: recursive
+      - name: Fetch most recent kokkos develop
+        run: |
+          cd externals/ekat/extern/kokkos
+          if ! git remote | grep -q upstream; then
+            git remote add upstream git@github.com:kokkos/kokkos.git
+          fi
+          git fetch upstream
+          git checkout develop
+          git reset --hard upstream/develop
+      - name: Show action trigger
+        uses: ./.github/actions/show-workflow-trigger
+      - name: Run tests
+        uses: ./.github/actions/test-all-eamxx
+        with:
+          build_type: ${{ matrix.build_type }}
+          machine: ghci-snl-cpu
+          generate: ${{ env.generate }}
+          submit: ${{ env.submit }}
+          cmake-configs: CMAKE_CXX_STANDARD=20;Kokkos_ENABLE_OPENMP=ON
```

Comment:
C++20. We could think about switching other testing to C++20, without updating kokkos.